openedx_proversity_reports/reports/time_spent_report_per_user.py

```diff
@@ -0,0 +1,249 @@
+        raise GoogleBigQueryInformationError('Google Service Account credentials or porject ID were not provided.')
+
+    credentials = service_account.Credentials.from_service_account_info(
+        service_account_credentials,
+        scopes=BIGQUERY_API_SCOPES,
+    )
+    google_bigquery_client = Client(
+        project=google_project_id,
+        credentials=credentials,
+    )
+
+    return google_bigquery_client
+
+
+def block_type_filter(block_item):
+    """
+    Return True if the block type does not exists in block_type_blacklist otherwise False.
+    This functionallity is to get only the component block types and discard the other block types.
+
+    Args:
+        block_item: Course block item to be matched.
+    """
+    block_type_blacklist = (
+        'course',
+        'chapter',
+        'sequential',
+        'vertical',
+    )
+    return True if block_item.block_type not in block_type_blacklist else False
+
+
+def get_google_bigquery_job_config():
+    """
+    Return the Google BigQuery job configuration.
+
+    Returns:
+        job_config: google.cloud.bigquery.job.QueryJobConfig instance.
+    """
+    job_config = bigquery.QueryJobConfig()
+    job_config.maximum_bytes_billed = getattr(settings, 'OPR_GOOGLE_BIGQUERY_MAX_PROCESS_BYTES', None)
+    job_config.use_query_cache = getattr(settings, 'OPR_GOOGLE_BIGQUERY_USE_CACHE', False)
+
+    return job_config
+
+
+def get_google_bigquery_query(course_dataset_name, date):
+    """
+    Return the Google BigQuery query for the time_on_asset_daily table.
+
+    Args:
+        course_dataset_name: Dataset name where the table is stored.
+        date: Date to filter the query. Date format: '%Y-%m-%d' e.g. '2019-01-01'
+    Returns:
+        query_string: The query string to make the query.
+    Raises:
+        GoogleBigQueryInformationError: If OPR_GOOGLE_CLOUD_PROJECT_ID or course_dataset_name
+                                        were not provided or are None.
+    """
+    google_project_id = getattr(settings, 'OPR_GOOGLE_CLOUD_PROJECT_ID', '')
+    query_max_result_number = getattr(settings, 'OPR_GOOGLE_BIGQUERY_MAX_NUMBER_RESULTS_PER_QUERY', 1000)
+
+    if not (google_project_id or course_dataset_name):
+        raise GoogleBigQueryInformationError('Google cloud project id or course_dataset_name are missing.')
+
+    query_string = """
+        SELECT course_id, username, module_id, time_umid5, time_umid30
+        FROM `{google_project_id}.{bigquery_dataset}.time_on_asset_daily`
+        WHERE module_id NOT LIKE '%vertical%'
+        AND module_id NOT LIKE '%sequential%'
+        AND time_umid5 IS NOT NULL
+        AND time_umid30 IS NOT NULL
+        AND PARSE_DATETIME('%Y-%m-%d', date) = '{query_date}' LIMIT {max_result_number}
+    """.format(
+        google_project_id=google_project_id,
+        bigquery_dataset=course_dataset_name,
+        query_date=date,
+        max_result_number=query_max_result_number,
+    )
+
+    return query_string
```

Comment:
merge with previous line

Reply:
No, let's keep it as it is, since is better to understand the function. @andrey-canon

Reply:
ok